core/sensors.py

```diff
@@ -147,6 +148,33 @@ def get_db_weather_data(session, date_from, date_to):
     result_df = DataFrame(session.execute(query).fetchall())
 
     if len(result_df.index) > 0:
-        result_df.set_index("timestamp", inplace=True)
+        result_df.set_index("timestamp", inplace=True) # set the index to become "timestamp" column
 
     return result_df
+
+
+def get_db_weather_forecast(session, date_from, date_to):
+    """
+    Returns weather forecast data (stored in the DB) for specific period of time as pandas data frame.
+
+    Arguments:
+        session: sqlalchemy active session object
+        date_from: date range from
+        date_to: date range to
+    Returns:
+        result_df: data frame containing sensor data for specific period of time
+    """
+
+    query = session.query(WeatherForecastsClass.timestamp,).filter(
```

Comment:
The trailing comma reminded me: We don't have a strict policy that code must be formatted using black (if you're not familiar: https://github.com/psf/black), but most of it is, and we'll occasionally run black over all the files. I format everything I write with black, Nick I think doesn't, although I'm not sure about that.

Reply:
So apparently the *weather_forecast* table **is logging** the time of the request (the column is called *time_created*). This is not actually the time of the request to the openweathermap API, but rather the time at which the row is created in the table - they should be very similar in any case (I actually did include `time_created = Column(DateTime(), server_default=func.now())` in the SQLAlchemy class).

What I don't understand is why the *iweather* table has Null entries for *time_created*, when it also has `Column(DateTime(), server_default=func.now())` in its SQLAlchemy class....

Not sure how you would want to deal with this, but given that the forecast time is labelled *timestamp* in *iweather* (and I am assuming the GES model is coded up to read this column from the table), I guess it makes sense to stick with this same convention in *weather_forecast*?